### PR TITLE
feature/NJP-29

### PR DESCRIPTION
**Pull request by Deideroh92**

### CODE REVIEW

File: api/routes/notes/get-note.routes.js [hunk 1]
```diff
@@ -167,4 +167,32 @@ router.delete(
...
 );
 
+// Retrieve note's revisions
+router.get('/users/:username/notes/:noteId', isAuthor, async (req, res) => {
+	const { data, error } = await supabase
+		.from('revisions')
+		.select('createdat, revision_id')
+		.eq('note', req.params.noteId);
+
+	if (error) throw new Error(error.message);
```
Comment (AmiralBl3ndic) [inline api/routes/notes/get-note.routes.js]: Can you directly throw the error if one occurs please chief? 🚀😄👨‍💻

File: api/routes/notes/get-note.routes.js [hunk 2]
```diff
@@ -167,4 +167,32 @@ router.delete(
...
+
+	for (id in data.revision_id) {
+		const { data: data2, error: error2 } = await supabase
+			.from('modifications')
+			.select('modification_id,position,previous,modified')
+			.eq('revision', id);
+		modification_data.push(data2[0]);
+	}
+
+	if (error2) throw new Error(error2.message);
```
Comment (AmiralBl3ndic) [inline api/routes/notes/get-note.routes.js]: Can you directly throw the error if one occurs please? 🙏💻

**Review by AmiralBl3ndic — CHANGES_REQUESTED** [review]

File: api/routes/notes/get-note.routes.js (hunk 2 above)
Comment (Deideroh92) [reply api/routes/notes/get-note.routes.js]: No problem chief

File: api/routes/notes/get-note.routes.js (hunk 1 above)
Comment (Deideroh92) [reply api/routes/notes/get-note.routes.js]: No problem chief

File: api/routes/notes/get-note.routes.js [hunk 3]
```diff
@@ -167,4 +167,32 @@ router.delete(
...
+	const { data, error } = await supabase
+		.from('revisions')
+		.select('createdat, revision_id')
+		.eq('note', req.params.noteId);
+
+	if (error) throw new Error(error.message);
+
+	var modification_data = [];
+
+	for (id in data.revision_id) {
```
Comment (AmiralBl3ndic) [inline api/routes/notes/get-note.routes.js]: You cannot iterate over that. `data` itself is an array.

**Review by AmiralBl3ndic — CHANGES_REQUESTED** [review]
Required changes told in the call

**Review by AmiralBl3ndic — APPROVED** [review]

**Pull request merged**